packages/backend-common/src/s3.ts

```diff
@@ -101,10 +123,26 @@ export const getObjectText = async (
 		for await (const chunk of body) {
 			chunks.push(chunk);
 		}
-		return Buffer.concat(chunks).toString('utf-8');
-	} catch (error) {
+		return {
+			status: S3Status.Success,
+			text: Buffer.concat(chunks).toString('utf-8'),
+		};
+	} catch (error: unknown) {
+		if (error instanceof Error) {
+			if (error.name === 'NoSuchKey') {
+				return {
+					status: S3Status.Failure,
+					statusCode: 410,
```

Comment:
non-blocking: `statusCode` isn't used in `api/src/index.ts` yet. Maybe it's better to keep it that way and remove `statusCode` from `GetObjectTextFailure`? That way it's clearer, reading `index.ts` which cases result in which status codes?